tests/mongodb/test_mongodb_source.py

```diff
@@ -356,3 +357,50 @@ def test_arrow_types(destination_name):
 
     info = pipeline.run(res, table_name="types_test")
     assert info.loads_ids != []
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_filter(destination_name):
+    """
+    The field `runtime` is not set in some movies,
+    thus incremental will not work. However, adding
+    an explicit filter_, which says to consider
+    only documents with `runtime`, makes it work.
+    """
+    pipeline = dlt.pipeline(
+        pipeline_name="mongodb_test",
+        destination=destination_name,
+        dataset_name="mongodb_test_data",
+        full_refresh=True,
+    )
+    movies = mongodb_collection(
+        collection="movies",
+        incremental=dlt.sources.incremental("runtime", initial_value=500),
+        filter_={"runtime": {"$exists": True}},
+    )
+    pipeline.run(movies)
+
+    table_counts = load_table_counts(pipeline, "movies")
+    assert table_counts["movies"] == 23
+
+
+@pytest.mark.parametrize("destination_name", ALL_DESTINATIONS)
+def test_filter_intersect(destination_name):
+    """
+    Check that using in the filter_ fields that
+    are used by incremental is not allowed.
+    """
+    pipeline = dlt.pipeline(
+        pipeline_name="mongodb_test",
+        destination=destination_name,
+        dataset_name="mongodb_test_data",
+        full_refresh=True,
+    )
+    movies = mongodb_collection(
+        collection="movies",
+        incremental=dlt.sources.incremental("runtime", initial_value=20),
+        filter_={"runtime": {"$gte": 20}},
```

Comment:
would it be worth adding tests for when $lts < $gte filter for the same field or thinking about whether we should handle this in some way?